data-platform/core-converged-db/oracle-database-23c-new-features/OracleJDBC23cNewFeatures/src/main/java/swierzy/oraclejdbc23newfeatures/Main.java

```diff
@@ -0,0 +1,156 @@
+        PreparedStatement psAux = conAux.prepareStatement("SELECT COUNT(*) FROM TEST");
+        // auxiliary ResultSet object used to check number of rows in TEST table
+        ResultSet rsAux;
+        int countInt;
+
+        /*
+           Streams Ingestion additionally to traditional JDBC parameters, like
+           connection string, username and password, requires providing the following data
+           1. ExecutorService: pool of threads responsible for execution
+           2. buffer size
+           3. interval (in seconds) between pushing the data into the database
+           4. names of target schema, table and columns
+         */
+
+        System.out.println("Main.streamsIngestDemo: connecting to the database.");
+        ExecutorService es = Executors.newFixedThreadPool(5);
+        ReactiveStreamsIngestion rs = ReactiveStreamsIngestion
+                .builder()
+                .url("<db_url>")
+                .username("<db_username>")
+                .password("<db_password>")
+                .executor(es)
+                .bufferRows(60)
+                .bufferInterval(Duration.ofSeconds(2))
+                .schema("<db_username>")
+                .table("TEST")
+                .columns(new String[] {"VAL"})
+                .build();
+        System.out.println("Main.streamsIngestiondemo: connected to the database");
+        // Streams Ingestion uses reactive calls and provides its own publisher and subscriber
+        PushPublisher<Object[]> pushPublisher = ReactiveStreamsIngestion.pushPublisher();
+        pushPublisher.subscribe(rs.subscriber());
+        System.out.println("Main.streamsIngestionDemo: ReactiveStreamsIngestion configured.");
+
+        for (int i = 1; i <= 60; i++) {
+            // generation of a random value
+            value = UUID.randomUUID().toString();
+            // pushing the data into the database
+            pushPublisher.accept(new Object[] {value});
+            System.out.println("Main.streamsIngestionDemo: A random string #"+i+" has been generated and pushed into the database stream.");
+
+            /*
+               Auxiliary PreparedStatement object got from the auxiliary, separate database connection,
+               checks the number of rows in the TEST table
+            */
+            rsAux = psAux.executeQuery();
+            rsAux.next();
+            countInt = rsAux.getInt(1);
+            System.out.println("Main.streamsIngestionDemo: Number of rows in TEST table: "+countInt);
+            Thread.sleep(200);
+        }
+
+        // closing the stream and the executor service
+        pushPublisher.close();
+        rs.close();
+        es.shutdown();
+        System.out.println("Main.streamsIngestionDemo: end of method and return to Main.main.");
+    }
+    public static void main(String[] args) {
+        try {
+            System.out.println("Main.main: begin of the demonstration");
+            Class.forName("oracle.jdbc.driver.OracleDriver");
```

Comment:
Very old way of using JDBC. Please check https://docs.oracle.com/en/database/oracle/oracle-database/21/jajdb/oracle/jdbc/datasource/impl/OracleDataSource.html for a modern way.